pyuvsim/simsetup.py

```diff
@@ -487,12 +496,15 @@ def initialize_catalog_from_params(obs_params, input_uv=None):
     return np.array(catalog), source_list_name
 
 
-def parse_telescope_params(tele_params, config_path):
+def parse_telescope_params(tele_params, config_path=''):
     """
     Parse the "telescope" section of obsparam.
```

Comment:
The main thing that concerns me about this function is that the `telescope_config` file also specifies the beam dictionary, which is used to assign beam models to antennas. I can't see where else (absent a telescope config file) the beam models are loaded and assigned.

Reply:
So if there isn't a telescope config then it returns empty lists and dicts for the beam info. The point there being that some users may want the antenna array functionality of this function without wanting to have to specify beam models (indeed like healvis for example!) In which case you can feed without a telescope config and it will just do the antenna array calculations. The empty beam info are still returned, though, in order to keep the output collection consistent.

Reply:
I think the beam setup should be included with telescope setup, personally, but if we're going to offer separate functions for layout+metadata and beams, then the option to get the beam list/dictionary should be removed from this function.

Reply:
Hmm, I think we may be confused. There is only one function to do telescope and beam setup: its `parse_telescope_params`. What I've done is enabled it to _only_ do the telescope setup if the beam information is not present, instead of failing, such that `healvis` etc. can use this function to do what they need to do without it failing. Does that make sense?

Reply:
I really think we want to keep the required beam functionality -- healvis requires a beam too, yes? If the issue is that healvis can only accept one beam, then there should be an error raised in healvis if multiple beams are specified.

Reply:
Hmm, well if the entire philosophy here is to make these functions more usable by packages outside of `pyuvsim`, then we will need to make these things more flexible, otherwise it will simply be much easier for other packages to roll their own simsetup functions (as we did in `healvis` to begin with because it was simply easier than trying to force `pyuvsim`'s adopted routines into `healvis`'s goals).

I think this is a fairly reasonable extension, keeping all functionality that `pyuvsim` requires but enabling it to be used in ways that other simulators might find useful, such as not needing to specify beam parameters when setting up a telescope array (which I think is a really reasonable thing to want to do!). So I guess I say, if we want `pyuvsim` routines to be more acccessible by other simulators, then no, we don't want to keep the required beam functionality. If you want `pyuvsim` specifically to retain that functionality, we can always do those checks in the wrapper functions that call `parse_telescope_params`...

Reply:
Thinking about this more, the problem seems to be that `parse_telescope_params` is serving two roles. The first is to interpret the standard obsparam YAML file keywords to extract required metadata for a UVData object, which of course doesn't require any beam information. The second is to set up all required telescope-related information for a simulation, which should include all beam models and their assignments to antennas. The way we've defined our setup yaml files necessarily intertwines these roles. The reason we defined these files, and deprecated simulating off of files, is that UVData doesn't provide sufficient information for a simulation, and we wanted to ensure that our setup standard would completely define a simulation.

I can accept this solution as a way of serving these two roles simultaneously, though I'm still a bit confused about the usage case for running this without paths to the config files in the dictionary. When run with no paths to external files, all this function appears to do is transform ENU antpos to ECEF. When run with reference to a layout file but not telescope_config, it has all the information needed to assign beams to antennas but no beams, so we're ignoring the "BeamIDs" column altogether. Overall, we're passing in a dictionary with all the required parameters for a UVData object and then getting back a slightly changed dictionary with the same information.